src/main/java/seedu/address/logic/commands/GroupCommand.java

```diff
@@ -0,0 +1,48 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.model.Model.COMPARATOR_GROUP_APPOINTMENTS;
+import static seedu.address.model.Model.COMPARATOR_GROUP_PATIENTS;
+
+import seedu.address.model.Model;
+
+/**
+ * Group all persons or appointments in the address book to the user.
+ */
+public class GroupCommand extends Command {
+
+    public static final String COMMAND_WORD = "group";
+
+    public static final String MESSAGE_SUCCESS_PATIENTS = "Grouped all patients.";
+    public static final String MESSAGE_SUCCESS_APPOINTMENTS = "Grouped all appointments.";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + " patients: Group all patients by their tags.\n"
+            + COMMAND_WORD + " appts: Group all appointments by their tags.\n";
+
+    private final String type;
+
+    public GroupCommand(String type) {
+        this.type = type;
+    }
+
+
+    @Override
+    public CommandResult execute(Model model) {
+        if (this.type.equals("patients")) {
+            requireNonNull(model);
+            model.updatePersonComparator(COMPARATOR_GROUP_PATIENTS);
+            return new CommandResult(MESSAGE_SUCCESS_PATIENTS);
+        } else {
+            requireNonNull(model);
+            model.updateAppointmentComparator(COMPARATOR_GROUP_APPOINTMENTS);
+            return new CommandResult(MESSAGE_SUCCESS_APPOINTMENTS);
+        }
```

Comment:
```suggestion
        requireNonNull(model);
        if (this.type.equals("patients")) {
            model.updatePersonComparator(COMPARATOR_GROUP_PATIENTS);
            return new CommandResult(MESSAGE_SUCCESS_PATIENTS);
        } else {
            model.updateAppointmentComparator(COMPARATOR_GROUP_APPOINTMENTS);
            return new CommandResult(MESSAGE_SUCCESS_APPOINTMENTS);
        }
```
Perhaps it might be better to put the null check statement outside the if else block, to reduce code duplication?